front-project/src/components/AccountMenu.jsx

```diff
@@ -0,0 +1,95 @@
+import * as React from "react";
+import Box from "@mui/material/Box";
+import Avatar from "@mui/material/Avatar";
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem";
+import ListItemIcon from "@mui/material/ListItemIcon";
+import Divider from "@mui/material/Divider";
+import IconButton from "@mui/material/IconButton";
+import Tooltip from "@mui/material/Tooltip";
+import Settings from "@mui/icons-material/Settings";
+import { getRandomColor } from "../utils/getRandomAccountColor";
+import { useSelector } from "react-redux";
+import { getUser } from "../features-store/auth/auth.selectors";
+
+export default function AccountMenu() {
+  const userName = useSelector(getUser).username;
```

Comment:
create selector getUsername
with getUser component will be rerendered every time any user field changes
with getUsername component will be rerendered only when username changes


https://react-redux.js.org/api/hooks
> When an action is dispatched, useSelector() will do a reference comparison of the previous selector result value and the current result value. If they are different, the component will be forced to re-render. If they are the same, the component will not re-render.